### ISSUE TITLE
`reportUnnecessaryCondition` rule

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#45]
```py
if True: # error: condition is always True
    ...
```

should not trigger on `defineConstant` values
